lib/phoenix_live_view/html_engine.ex

```diff
@@ -0,0 +1,398 @@
+defmodule Phoenix.LiveView.HTMLEngine do
+  @moduledoc """
+  The HTMLEngine that powers `.heex` templates and the `~H` sigil.
+  """
+
+  alias Phoenix.LiveView.HTMLTokenizer
+
+  @behaviour Phoenix.Template.Engine
+
+  # TODO: Use @impl true instead of @doc false when we require Elixir v1.12
+
+  @doc false
+  def compile(path, _name) do
+    trim = Application.get_env(:phoenix, :trim_on_html_eex_engine, true)
```

Comment:
```suggestion
    trim = Application.compile_env(:phoenix, :trim_on_html_eex_engine, true)
```

Reply:
Compile env can only be called in the module body and we don’t want to do that here because then you need to recompile LiveView when that config changes. It is currently not possible to use compile_env to express compile config across deps.

Reply:
👍🏼